node.go

```diff
@@ -25,15 +25,21 @@ type NodeOption func(*node)
 // WithTagsOfNode replaces tags of a node
 func WithTagsOfNode(tags ...string) NodeOption {
 	return func(modifiedNode *node) {
-		modifiedNode.tags = tags
+		tagsM := make(map[string]struct{})
+		for _, tag := range tags {
+			tagsM[tag] = struct{}{}
```

Comment:
@smcheema I think if you predefine struct {}{} something like var exists or something  
Or add a comment here so its easier to understand what struct {}{} here means

Reply:
@musanimu When we build a new struct for example struct `Apple` we do it as below:
```GoLang
type Apple struct {
	a int
}

func TestApple(t *testing.T) {
	apple:= Apple{a: 32}
}
```
`apple:= Apple{a: 32}` is equivalent to `apple:= (Apple)(struct {a int}{a: 32})` 

So when you ask what `struct{}{}` means, it is just creating an anonymous struct `struct{}`